### ISSUE TITLE
Replace all on-tap with on-click

### ISSUE THREAD

**Issue by balloob** [home-assistant/home-assistant-polymer#916]
@NovapaX in #911 linked to the [polymer docs](https://www.polymer-project.org/2.0/docs/devguide/gesture-events) describing that `on-tap` is no longer needed. We should update our components.

```
› git grep "on-tap" | wc -l
     145
```

**Comment by NovapaX** [comment]
yes... although in my experience both events have a different technical implementation. 'tap' is a polymer generated event (kind of a wrapper around the native MouseEvent), 'click' is a native browser event.

Doesn't really matter for most of the time. Unless event.detail is called in the bound function, because `tap` does have `event.detail` while the native `click` mouseEvent does not.

**Comment by balloob** [comment]
I think that we only use `event.detail` if we listen inside a dom-repeat and want to access the item. According to [the docs](https://www.polymer-project.org/2.0/docs/devguide/templates#handling-events), that is still supported.

**Comment by balloob** [comment]
Oh, reading the docs help. Did know we can do `e.model.set('property', 'value')`, nice.

**Comment by NovapaX** [comment]
> I think that we only use event.detail if we listen inside a dom-repeat and want to access the item.

yes, and to get xy coordinates.

`event.detail.x` on `tap` vs. `event.x` on `click`
